### PR TITLE
5622 state semantics

### PR DESCRIPTION
**Pull request by lzach83**
## Link to Issue
Closes: #5622 

## Description of Changes
- Changed instances of `chain` and `chainId` to `community` and `communityId`
- updating everything within the `/state` folder

## Test Plan
- Removed the `chainId` param from all `[CRUD]ThreadLinks` they were not being used within the function directly -- make sure all instances work
- CRUD Comments
- CRUD Groups 
- Searching Profiles
- CRUD Threads
- CRUD topics

### CODE REVIEW

**Review by masvelio — APPROVED** [review]
two small comments, other than that lgtm, but please resolve the confclits

File: packages/commonwealth/client/scripts/views/components/NewThreadForm/NewThreadForm.tsx [hunk 1]
```diff
@@ -153,7 +153,7 @@ export const NewThreadForm = () => {
         stage: app.chain.meta.customStages
           ? parseCustomStages(app.chain.meta.customStages)[0]
           : ThreadStage.Discussion,
-        chainId: app.activeChainId(),
+        communityId: app.activeChainId(),
```
[the remark this answers is not in the document]
Comment (lzach83) [reply packages/commonwealth/client/scripts/views/components/NewThreadForm/NewThreadForm.tsx:156]: Yes there is, @timolegros and I are going to take care of that once gating get's out the door.

Added a ticket to track this: https://github.com/hicommonwealth/commonwealth/issues/5753

**Comment by lzach83** [comment]
Converting to draft, i'm going to break this PR down. 

Will eventually close this pr

**Comment by lzach83** [comment]
Closing since all the separate parts have been merged

**Pull request closed without merging**